### PR TITLE
makefile: add extensive checks

### PR DESCRIPTION
**Pull request by gregwebs**
## What have you changed? (required)

Add more extensive static checks. Switch to the revive linter which is fast and configurable.

## What are the type of the changes (required)?

This is a tooling improvement. There was a bug in the usage of `TrimLeft`, but that only existed in test code. Also some ignored errors in the CLI and test.

## How has this PR been tested (required)?

`make check && make test`

### CODE REVIEW

File: pdctl/command/region_command.go [hunk 1]
```diff
@@ -70,8 +71,7 @@ func NewRegionCommand() *cobra.Command {
 }
 
 func showRegionCommandFunc(cmd *cobra.Command, args []string) {
-	var prefix string
-	prefix = regionsPrefix
+	var prefix = regionsPrefix
```
Comment (rleungx) [inline pdctl/command/region_command.go]: Maybe use `:=` ?

File: revive.toml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[rule.exported]
+[rule.if-return]
+[rule.var-naming]
+[rule.package-comments]
+[rule.range]
+[rule.receiver-naming]
+[rule.indent-error-flow]
+[rule.empty-block]
+[rule.superfluous-else]
+[rule.modifies-parameter]
```
Comment (rleungx) [inline revive.toml]: Maybe `increment-decrement` should be added?

File: pdctl/command/region_command.go (hunk 1 above)
Comment (gregwebs) [reply pdctl/command/region_command.go]: That seems like a linter rule :)

File: revive.toml (hunk 2 above)
Comment (gregwebs) [reply revive.toml]: I don't see the point to `++` in go. Its something additional to learn even though it is no different than `+=1` which is already quite concise. I would

File: pdctl/command/global.go [hunk 3]
```diff
@@ -122,6 +121,8 @@ func printResponseError(r *http.Response) {
 	io.Copy(os.Stdout, r.Body)
 }
 
+/* unused
```
Comment (nolouch) [inline pdctl/command/global.go]: please clean the unused code.

Comment (gregwebs) [reply pdctl/command/global.go]: I am concerned that `testCheckStores` is not being run by accident. I would like for @disksing to confirm that code should be removed (in this PR or send a new PR later).

Comment (gregwebs) [reply pdctl/command/global.go]: Okay, the code is removed now.

File: cmd/pd-tso-bench/main.go [hunk 4]
```diff
@@ -29,9 +29,9 @@ import (
 )
 
 var (
+	// sleep       = flag.Duration("sleep", time.Millisecond, "sleep time after a request, used to adjust pressure")
```
Comment (nolouch) [inline cmd/pd-tso-bench/main.go]: unused?

File: Makefile [hunk 5]
```diff
@@ -42,16 +43,49 @@ test:
...
-	@echo "vet"
-	@ go tool vet . 2>&1 | $(GOCHECKER)
-	@ go tool vet --shadow . 2>&1 | $(GOCHECKER)
-	@echo "golint"
-	@ golint ./... 2>&1 | $(GOCHECKER)
-	@echo "gofmt"
-	@ gofmt -s -l . 2>&1 | $(GOCHECKER)
+# setup needs to be ran just once to install dependencies
+setup:
+	go get github.com/twitchtv/retool
```
Comment (disksing) [inline Makefile]: We can move it into `tool-install`. `go get` will know it is already installed and skip the download&build process.

Comment (gregwebs) [reply Makefile]: I was trying to differentiate between global tools and those managed by retool. Right now on CI `setup` is ran  but `tool-install` is not. Even though `tool-install` is idempotent, it takes up time.

**Comment by disksing** [comment]
failure due to leader election too slow. Re-run.

**Pull request merged**